### PR TITLE
Feature asp retrieval

### PR DESCRIPTION
**Pull request by quinten-goens**
**Changes:** 

- Add four new airspace geometry retrieval functions (acc_sf, ansp_sf, es_sf, fir_sf) that query Oracle spatial tables and return sf objects for ACC/OAC, ANSP, Elementary Sector, and FIR airspaces by CFMU AIRAC cycle. Based on the queries used in pruatlas.
- Add sf as a suggested dependency.
- Add Quinten Goens as a package co-author.

**Request:**
Could you kindly review and merge to main?

### CODE REVIEW

File: .gitignore
Comment (espinielli) [inline .gitignore:1]: This is not really needed...maybe better to add the tests? (But i think I had none, sigh!)

File: R/airspaces.R [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+#' @description
+#' Returns ACC (Area Control Centre) and OAC (Oceanic Area Control Centre)
+#' airspace geometries as an `sf` object for the specified CFMU AIRAC cycle.
+#'
+#' # Note
+#' You need to either provide a connection `conn` that has access to
+#' `ENV_SP.AIRSPACE`, `PRU_STAT_AUA` and `SWH_MAP` or go with the default
+#' which uses PRU_READ to establish a [db_connection()].
+#'
+#' The `sf` package is required but only suggested; install it if not available.
```
Comment (espinielli) [inline R/airspaces.R:12]: Why? Is this because it follows ROracle and the moment you need to query the DB, then maybe you need the GIS parts...?
Probably ok, like it is

File: R/airspaces.R [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+#'
+#' @inheritParams airlines_tbl
+#'
+#' @param cfmu_airac the AIRAC cycle number (CFMU format), e.g. `"517"`
```
Comment (espinielli) [inline R/airspaces.R]: could you please put a link to the `airac` package?
What do you think?

File: R/airspaces.R [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+#' * `NAME`: the airspace name
+#' * `CODE`: the airspace code
+#' * `AIRSPACE_TYPE`: the airspace type (`ACC` or `OAC`)
+#' * `geometry`: the airspace polygon geometry
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' acc <- acc_sf(cfmu_airac = "517")
```
Comment (espinielli) [inline R/airspaces.R:37]: Does it also work with integer/number? Like 517L or 517

File: R/airspaces.R [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+      AND A.SHAPE IS NOT NULL
+  "
+
+  query <- DBI::sqlInterpolate(conn, sqlq, CFMU_AIRAC = cfmu_airac)
+  res <- DBI::dbSendQuery(conn, query)
+  data <- DBI::fetch(res, n = -1)
+  DBI::dbClearResult(res)
+
+  data |>
+    dplyr::first() |>
```
Comment (espinielli) [inline R/airspaces.R:110]: Why this? Does the query return multiple geometries?  Then it is a wrong one...

**Review by espinielli — COMMENTED** [review]
I left some comments and "why" questions for your evaluation.
Nothing blocks the merge IMHO (also we can fix things later) but it would be good to know the "why"s.
Thanks

File: .gitignore
Comment (quinten-goens) [reply .gitignore:1]: It was not a full test, just me trying out whether the functions work and a visual inspection of the returned airspace definitions.

File: R/airspaces.R (hunk 1 above)
Comment (quinten-goens) [reply R/airspaces.R:12]: There are some sf functions used in the airspace.R for postprocessing after the query (as per your code in the original pruatlas).

File: R/airspaces.R (hunk 2 above)
Comment (quinten-goens) [reply R/airspaces.R]: I'll add it.

File: R/airspaces.R (hunk 3 above)
Comment (quinten-goens) [reply R/airspaces.R:37]: Nope. Will add compatibility.

File: .gitignore
Comment (quinten-goens) [reply .gitignore:1]: So it was not a testthat:: test.

File: R/airspaces.R (hunk 4 above)
Comment (quinten-goens) [reply R/airspaces.R:110]: I copied your original code: https://github.com/euctrl-pru/pruatlas/blob/master/data-raw/export_acc_geojson.R - I am not sure why you added this but I am sure you had a good reason.

File: R/airspaces.R (hunk 3 above)
Comment (quinten-goens) [reply R/airspaces.R:37]: Added, see https://github.com/eurocontrol/eurocontrol/commit/37ca068d12e30d3d8e61da6e253fcf0c2f0d91d6

File: R/airspaces.R (hunk 1 above)
Comment (quinten-goens) [reply R/airspaces.R:12]: Note: It's not lazily loaded these functions.

File: R/airspaces.R (hunk 2 above)
Comment (quinten-goens) [reply R/airspaces.R]: Added, see https://github.com/eurocontrol/eurocontrol/commit/37ca068d12e30d3d8e61da6e253fcf0c2f0d91d6

File: .gitignore
Comment (espinielli) [reply .gitignore:1]: then the file is gone, and remove from .gitignore...

Comment (quinten-goens) [reply .gitignore:1]: Removed the line from .gitignore: https://github.com/eurocontrol/eurocontrol/commit/cda8bfe9fb0deae1850e9d79e3dd3b01ff266754

File: R/airspaces.R (hunk 4 above)
Comment (espinielli) [reply R/airspaces.R:110]: archeology... 🤭

**Comment by espinielli** [comment]
go ahead and merge/publish.
Thanks!
